src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    const sizes = [a, b, c].sort((szA: number, szB: number) => szB - szA);
 
+    if (sizes.some((size: number) => size <= 0)
+      || sizes[0] >= sizes[1] + sizes[2]
+    ) {
+      throw new Error('Unexpected error');
```

Comment:
'unexpected error' is not clear. specify another message, for ex. 'invalid triangle sizes'